packages/material-ui/src/SwipeableDrawer/SwipeableDrawer.js

```diff
@@ -284,26 +284,36 @@ class SwipeableDrawer extends React.Component {
           ? window.innerHeight - event.changedTouches[0].clientY
           : event.changedTouches[0].clientY;
     }
-    const translateRatio = this.getTranslate(current) / this.getMaxTranslate();
-
-    // We have to open or close after setting swiping to null,
-    // because only then CSS transition is enabled.
-    if (translateRatio > 0.5) {
-      if (this.isSwiping && !this.props.open) {
-        // Reset the position, the swipe was aborted.
-        this.setPosition(this.getMaxTranslate(), {
-          mode: 'enter',
+
+    // get the beginning of the swipe, pending on orientation
+    const beginSwipe = isHorizontal(this.props) ? this.startX : this.startY;
+
+    // compute minimal swipe distance by hysteresis property
+    const minDistance = this.getMaxTranslate() * this.props.hysteresis;
+
+    // get the real distance
+    const distance = Math.abs(beginSwipe - current);
+
+    if (beginSwipe > current) {
+      // closing swipe
+      if (distance < minDistance) {
+        // distance too short, go back to opened state
+        this.setPosition(0, {
+          mode: 'exit',
         });
       } else {
         this.props.onClose();
       }
-    } else if (this.isSwiping && !this.props.open) {
-      this.props.onOpen();
     } else {
-      // Reset the position, the swipe was aborted.
-      this.setPosition(0, {
-        mode: 'exit',
-      });
+      // opening swipe
+      if (distance < minDistance) {
+        // distance too short, go back to closed state
+        this.setPosition(_this.getMaxTranslate(), {
```

Comment:
Where does `_this` come from?

Reply:
I tested it in the npm package - it uses "_this". I forgot to change it afterwards.